### ISSUE TITLE
[BUG] Proxy option for feed

### ISSUE THREAD

**Issue by squromiv** [marienfressinaud/FreshRSS#3975]
**Describe the bug**
I tried to set several HTTP proxies in properties of feed - [Web Filtering Proxy](https://webproxy.diladele.com/docs/), [SQUID](http://www.squid-cache.org/), [HandyCache](http://handycache.ru/component/option,com_remository/Itemid,2/func,startdown/id,66/)

![1](https://user-images.githubusercontent.com/15985877/141530024-72b3f4f4-af1b-4d4b-9123-274ac95b4888.png)

But FreshRSS fails to work with all of them.
Particular errors, seen in FreshRSS:

![3](https://user-images.githubusercontent.com/15985877/141530402-a2d5b712-9da2-4dcd-8e2b-a272cdb65abc.png)

![2](https://user-images.githubusercontent.com/15985877/141530495-0996e0e5-5708-4ab5-a76f-b16935ccdc00.png)
Log-messages from proxy apps:
![4](https://user-images.githubusercontent.com/15985877/141531498-471c1b9f-4d61-47aa-8aee-a8189a4b0ca0.png)

![6](https://user-images.githubusercontent.com/15985877/141532094-4b5bf6bc-6e1c-4dd9-b3b2-5ede83a65cf9.png)

The [feed](https://github.com/FreshRSS/FreshRSS/releases.atom) is accessible in browser through proxy. Also it is normally updated in FreshRSS, if proxy is turned off in properties.

**Environment information:**
- OS: Windows Server2016x64
- Browser: Firefox 80+
- FreshRSS version: 1.19.0-dev 
- Database version: SQLite
- PHP version: PHP 7.4
- Installation type: OpenServer

**Comment by printfuck** [comment]
I think I found the issue in:
https://github.com/FreshRSS/FreshRSS/blob/a5410f6d3748c57fc02da33eef842f993b8bc251/lib/lib_rss.php#L191-L195

I replaced it with:
```
        if (!empty($attributes['curl_params']) && is_array($attributes['curl_params'])) {
                array_merge($attributes['curl_params'], $curl_options);
        }
```
And it seemed to work again. Can someone verify that?

**Comment by math-GH** [comment]
@squromiv is this bug still relevant and reproduce able with V1.19.2 and V1.20.0-dev?

**Comment by squromiv** [comment]
Will check.

**Comment by squromiv** [comment]
1.20.0-dev
Have tested with another app [CCProxy](https://www.youngzsoft.net/ccproxy/). It fails. 
Does anyone know a  proxy definitely working with FreshRSS? It looks like a fake feature this day.

**Comment by printfuck** [comment]
Hi there. I am using the 1.20-dev with sockd/danted socks5 proxy and it works flawlessly. I think it might be related to http proxies.

**Comment by Alkarex** [comment]
@squromiv Please try with a `curl` command first.
FreshRSS is using cURL.

**Comment by printfuck** [comment]
I remember that there actually have been issues with fetching feeds via proxy, when @squromiv initially opened this issue. At this time I had used the the docker :latest image and switched back to the most recent release. Later I went back to :latest and now :edge, as I found out, that it worked again.

I've just setup a tinyproxy http proxy server and tested it with freshrss:edge (which is 1.20.0-dev, I guess?) and it also worked flawlessly.
